fvm/evm/emulator/emulator.go

```diff
@@ -573,6 +573,13 @@ func (proc *procedure) run(
 		res.GasConsumed = execResult.UsedGas
 		res.GasRefund = proc.state.GetRefund()
 		res.Index = uint16(txIndex)
+
+		if proc.extraPrecompiledIsCalled() {
```

Comment:
do we need this if, couldn't we just always call `proc.capturePrecompiledCalls()` and in case it isn't called it would return nil

Reply:
nit: should the type `ExtraPrecompiles` just have encode function and then this can become `res,PrecompiledCalls, err = proc.config.ExtraPrecompiles.Encode()` and we don't have to define these bellow methods on the procedure?

Reply:
same for reset, this methods [feel like they have feature envy](https://refactoring.guru/smells/feature-envy)

Reply:
@sideninja going to merge this PR and open up a follow up with refactored code.